packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,448 @@
+
+    if (targetId && event.target.hasAttribute('data-terra-pill')) {
+      setTabIndex('-1');
+      currentPill.current = targetId;
+      focusNode.current = pills.findIndex((element) => element.id === targetId);
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const handleOnSelectRollUp = (event) => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    if (isCollapsible && isCollapsed) {
+      if (event.type === 'keydown') {
+        isRollUpRemoved.current = true;
+        focusNode.current = pills.length;
+      } else {
+        setTabIndex('-1');
+        focusNode.current = pills.length;
+      }
+    } else {
+      setShowRollupPillInteraction(false);
+    }
+    setIsCollapsed(!isCollapsed);
+  };
+
+  const handlePillListOnblur = () => setContainerTabindex('-1');
+
+  const filterPillsProps = {};
+  filterPillsProps.onKeyDown = handlePillListKeyDown;
+  filterPillsProps.onBlur = handlePillListOnblur;
+
+  const pillListClassNames = classNames(
+    cx([
+      'pills-group',
+      theme.className,
+    ]),
+    customProps.className,
+  );
+
+  const pillInteractionHintID = 'terra-pills-interaction-hint';
+  const pillGroupInteractionHintID = 'terra-pills-group-interaction-hint';
+  let removedPillInteractionHint = '';
+  const pillGroupAriaDescribedBy = ariaDescribedBy ? `${ariaDescribedBy} ${pillGroupInteractionHintID}` : pillGroupInteractionHintID;
+  let pillGroupInteractionHint = intl.formatMessage({ id: 'Terra.pills.hint.pillList' }, { numberOfPills: React.Children.count(children) });
+  if (isCollapsible && (rollUpCount > 0) && isCollapsed) {
+    pillGroupInteractionHint += `, ${intl.formatMessage({ id: 'Terra.pills.hint.rollupNotVisible' }, { pillsNotVisibleCount: rollUpCount })}`;
+  } else if (isCollapsible && !isCollapsed && showRollupPillInteraction) {
+    removedPillInteractionHint = 'hidden items are displayed now...'; // needs transtaions lp052179
+  }
+  if (pillRemoved) {
+    removedPillInteractionHint = `${removedLabel.current} was removed...`; // needs translations lp052179
+  }
+
+  const renderChildren = (items) => {
+    const pills = React.Children.map(items, (pill) => {
+      if (React.isValidElement(pill)) {
+        return React.cloneElement(pill, {
+          onRemove: handleOnRemove, onSelect: handleOnPillSelect, pillInteractionHintID,
+        });
+      }
+      return undefined;
+    });
+
+    const reducedArray = pills.splice(0, updatedCount);
+
+    return reducedArray;
+  };
+
+  return (
+    <div
+      {...customProps}
+      {...filterPillsProps}
+      aria-label={!ariaLabelledBy ? ariaLabel : undefined}
+      aria-labelledby={ariaLabelledBy}
+      aria-describedby={pillGroupAriaDescribedBy}
+      className={pillListClassNames}
+      ref={filterPillsRef}
+      role="list"
+      tabIndex={containerTabindex}
+    >
+      <VisuallyHiddenText
+        aria-live="assertive"
+        id={pillInteractionHintID}
```

Comment:
If multiple filter pillls are rendered on a page, we could end up have multiple VisuallyHiddenText with the same id. We should use `uuid` to generate a unique one every time.